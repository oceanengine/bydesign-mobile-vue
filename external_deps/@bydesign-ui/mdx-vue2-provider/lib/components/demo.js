/**
 * Component Demo
 */
import createBEM from '../utils/bem';
import HL from './highlight';
import { openCodepen, openSandbox, copyToClipboard } from '../utils/editor';
var bem = createBEM('demo');

var Demo = function Demo() {
  return {
    props: {
      title: [String, Number],
      desc: [String, Number],
      code: String,
      src: String,
      srcRender: Function,
      maxHeight: Number
    },
    data: function data() {
      return {
        unfolded: false
      };
    },
    methods: {
      setUnfolded: function setUnfolded(value) {
        this.unfolded = value;
      }
    },
    render: function render() {
      var h = arguments[0];
      var title = this.title,
          desc = this.desc,
          code = this.code,
          srcRender = this.srcRender,
          unfolded = this.unfolded,
          setUnfolded = this.setUnfolded,
          _this$maxHeight = this.maxHeight,
          maxHeight = _this$maxHeight === void 0 ? 200 : _this$maxHeight;
      var containerStyle = {
        maxHeight: "".concat(maxHeight, "px")
      };
      return h("div", {
        "class": bem()
      }, [h("div", {
        "class": bem('title'),
        "attrs": {
          "id": title.toString()
        }
      }, [title]), srcRender !== undefined && h("div", {
        "class": bem('box')
      }, [h("div", {
        "class": bem('show')
      }, [srcRender()]), desc && h("div", {
        "class": bem('desc')
      }, [desc]), code !== undefined && h("div", {
        "class": bem('code-tool')
      }, [h("div", {
        "class": bem('operations')
      }, [h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openCodepen(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/codepen.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openSandbox(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/sandbox.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            copyToClipboard(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/copy.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            setUnfolded(!unfolded);
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/code.svg')
        }
      })])]), code !== undefined && unfolded && h("div", {
        "class": bem('code'),
        "style": containerStyle
      }, [h(HL, {
        "attrs": {
          "code": unescape(code)
        }
      })])])]);
    }
  };
};

export default Demo;