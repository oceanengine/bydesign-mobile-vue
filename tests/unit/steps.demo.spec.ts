import Demo from '@src/components/steps/demo/index.vue';
import { snapshotDemo } from '../demo';

snapshotDemo(Demo);
