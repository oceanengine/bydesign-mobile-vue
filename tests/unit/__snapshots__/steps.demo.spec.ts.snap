// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div class="demo steps">
  <div class="byted-header--wrapper">
    <div class="byted-header byted-header--default byted-header--fixed byted-hairline--bottom">
      <div class="byted-header__left"><i class="byted-header__icon" style="width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title"><span class="demo-header-icon"><img src="https://s2.pstatp.com/cdn/expire-1-d/byted-creative-app/static/ByDesign-logo.svg" class="img-icon"></span></div>
    </div>
  </div>
  <div><span class="byted-icon byted-icon-module-search" style="font-size: 20px;"><svg width="20" height="20" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M60 43V21H4V43H60Z" fill="transparent" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M52 31C52 32.1046 51.5523 33.1046 50.8284 33.8284C50.1046 34.5523 49.1046 35 48 35C45.7909 35 44 33.2091 44 31C44 28.7909 45.7909 27 48 27C50.2091 27 52 28.7909 52 31Z" fill="transparent"></path><path d="M54 37L50.8284 33.8284M50.8284 33.8284C51.5523 33.1046 52 32.1046 52 31C52 28.7909 50.2091 27 48 27C45.7909 27 44 28.7909 44 31C44 33.2091 45.7909 35 48 35C49.1046 35 50.1046 34.5523 50.8284 33.8284Z" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path></svg></span></div>
  <div>
    <div class="byted-steps byted-steps--horizontal byted-steps--label-horizontal" style="margin-bottom: 24px;">
      <div class="byted-step byted-step--horizontal byted-step--md byted-step--finished">
        <div class="byted-step__head"><i class="byted-icon byted-icon--check-one byted-step__icon">test-file-stub</i><span class="byted-step__title">已完成</span></div>
        <div class="byted-step__split-panel"></div>
      </div>
      <div class="byted-step byted-step--horizontal byted-step--md byted-step--inProgress byted-step--active">
        <div class="byted-step__head"><span class="byted-step__icon-number">2</span><span class="byted-step__title">进行中</span></div>
        <div class="byted-step__split-panel"></div>
      </div>
      <div class="byted-step byted-step--horizontal byted-step--md byted-step--waiting">
        <div class="byted-step__head"><span class="byted-step__icon-number">3</span><span class="byted-step__title">待完成</span></div>
        <div class="byted-step__split-panel"></div>
      </div>
      <div class="byted-step byted-step--horizontal byted-step--md byted-step--error">
        <div class="byted-step__head"><span class="byted-step__icon-number">4</span><span class="byted-step__title">未完成</span></div>
      </div>
    </div>
    <div class="byted-steps byted-steps--horizontal byted-steps--label-vertical">
      <div class="byted-step byted-step--horizontal byted-step--md byted-step--finished">
        <div class="byted-step__head"><i class="byted-icon byted-icon--check-one byted-step__icon">test-file-stub</i>
          <div class="byted-step__split-panel"></div>
        </div><span class="byted-step__title">已完成</span>
      </div>
      <div class="byted-step byted-step--horizontal byted-step--md byted-step--inProgress byted-step--active">
        <div class="byted-step__head"><span class="byted-step__icon-number">2</span>
          <div class="byted-step__split-panel"></div>
        </div><span class="byted-step__title">进行中</span>
      </div>
      <div class="byted-step byted-step--horizontal byted-step--md byted-step--waiting">
        <div class="byted-step__head"><span class="byted-step__icon-number">3</span>
          <div class="byted-step__split-panel"></div>
        </div><span class="byted-step__title">待完成</span>
      </div>
      <div class="byted-step byted-step--horizontal byted-step--md byted-step--error">
        <div class="byted-step__head"><span class="byted-step__icon-number">4</span></div><span class="byted-step__title">未完成</span>
      </div>
    </div>
  </div>
  <div>
    <div class="steps-vertical-layout">
      <div class="byted-steps byted-steps--vertical byted-steps--label-horizontal" style="height: 300px; margin-right: 100px;">
        <div class="byted-step byted-step--vertical byted-step--md byted-step--finished">
          <div class="byted-step__head"><i class="byted-icon byted-icon--check-one byted-step__icon">test-file-stub</i><span class="byted-step__title">已完成</span></div>
          <div class="byted-step__split-panel"></div>
        </div>
        <div class="byted-step byted-step--vertical byted-step--md byted-step--inProgress byted-step--active">
          <div class="byted-step__head"><span class="byted-step__icon-number">2</span><span class="byted-step__title">进行中</span></div>
          <div class="byted-step__split-panel"></div>
        </div>
        <div class="byted-step byted-step--vertical byted-step--md byted-step--waiting">
          <div class="byted-step__head"><span class="byted-step__icon-number">3</span><span class="byted-step__title">待完成</span></div>
          <div class="byted-step__split-panel"></div>
        </div>
        <div class="byted-step byted-step--vertical byted-step--md byted-step--error">
          <div class="byted-step__head"><span class="byted-step__icon-number">4</span><span class="byted-step__title">未完成</span></div>
        </div>
      </div>
      <div class="byted-steps byted-steps--vertical byted-steps--label-vertical" style="height: 300px;">
        <div class="byted-step byted-step--vertical byted-step--md byted-step--finished">
          <div class="byted-step__head"><i class="byted-icon byted-icon--check-one byted-step__icon">test-file-stub</i><span class="byted-step__title">已完成</span></div>
          <div class="byted-step__split-panel"></div>
        </div>
        <div class="byted-step byted-step--vertical byted-step--md byted-step--inProgress byted-step--active">
          <div class="byted-step__head"><span class="byted-step__icon-number">2</span><span class="byted-step__title">进行中</span></div>
          <div class="byted-step__split-panel"></div>
        </div>
        <div class="byted-step byted-step--vertical byted-step--md byted-step--waiting">
          <div class="byted-step__head"><span class="byted-step__icon-number">3</span><span class="byted-step__title">待完成</span></div>
          <div class="byted-step__split-panel"></div>
        </div>
        <div class="byted-step byted-step--vertical byted-step--md byted-step--error">
          <div class="byted-step__head"><span class="byted-step__icon-number">4</span><span class="byted-step__title">未完成</span></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
